tests/jobs/test_scheduler_job.py

```diff
@@ -4769,6 +4773,375 @@ def test_max_active_runs_in_a_dag_doesnt_stop_running_dagruns_in_otherdags(self,
         assert dag1_running_count == 1
         assert running_count == 11
 
+    def test_dag1_is_mutated(self, dag_maker, session):
+        with dag_maker(
+            "test_dag1",
+            start_date=DEFAULT_DATE,
+            schedule=timedelta(days=1),
+            max_active_runs=1,
+        ) as dag1:
+            EmptyOperator(task_id="mytask")
+
+        with dag_maker(
+            "test_dag2",
+            start_date=timezone.datetime(2020, 1, 1),
+            schedule=timedelta(days=1),
+        ):
+            EmptyOperator(task_id="mytask")
+        assert dag1.dag_id == "test_dag2"
+
+    def test_max_active_runs_in_a_dag_doesnt_prevent_backfill_from_running(self, dag_maker):
+        session = settings.Session()
+        with dag_maker(
+            "test_dag1",
+            start_date=DEFAULT_DATE,
+            schedule=timedelta(days=1),
+            max_active_runs=1,
+        ) as dag:
+            EmptyOperator(task_id="mytask")
+        dag1_dag_id = dag.dag_id
+        dr = dag_maker.create_dagrun(run_type=DagRunType.SCHEDULED, state=State.QUEUED)
+        for _ in range(29):
+            dr = dag_maker.create_dagrun_after(dr, run_type=DagRunType.SCHEDULED, state=State.QUEUED)
+
+        with dag_maker(
+            "test_dag2",
+            start_date=timezone.datetime(2020, 1, 1),
+            schedule=timedelta(days=1),
+        ):
+            EmptyOperator(task_id="mytask")
+
+        dr = dag_maker.create_dagrun(run_type=DagRunType.SCHEDULED, state=State.QUEUED)
+        for _ in range(9):
+            dr = dag_maker.create_dagrun_after(dr, run_type=DagRunType.SCHEDULED, state=State.QUEUED)
+
+        scheduler_job = Job()
+        self.job_runner = SchedulerJobRunner(job=scheduler_job, subdir=os.devnull)
+
+        scheduler_job.executor = MockExecutor(do_update=False)
+        self.job_runner.processor_agent = mock.MagicMock(spec=DagFileProcessorAgent)
+
+        self.job_runner._start_queued_dagruns(session)
+        session.flush()
+        self.job_runner._start_queued_dagruns(session)
+        session.flush()
```

Comment:
Why do we need to do it twice? Just the volume of runs?

Reply:
yeah i think so. based this test off of one from ephraim.  but pretty sure that's the reason.  there are 30+ runs created and max runs default is 20